### PR TITLE
Object to context : add error to problems instead of throwing.

### LINKED ISSUE #958: Parser [patch] - handle thrown errors more gracefully
We currently throw an error on any unexpected input.

### PR DESCRIPTION
**Pull request by PrashanthCorp**
Fixes #958

### CODE REVIEW

File: src/mongo/MongoScrapbook.ts [hunk 1]
```diff
@@ -281,7 +281,9 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 		} else if (child instanceof ErrorNode) {
 			return {};
 		} else {
-			throw new Error("Unrecognized node type encountered");
+			let err = new Error(`Unrecognized node type encountered. We could not parse ${child.text}`);
```
Comment (EricJizbaMSFT) [inline src/mongo/MongoScrapbook.ts]: I don't think this is the right way to go about this. If you're call `addErrorToCommand` each time individually then you're asking for another bug like this to happen. Someone else could easily forget to do this in the future.

Plus the current code means we don't get any telemetry.

Comment (StephenWeatherford) [reply src/mongo/MongoScrapbook.ts]: This was meant only for a quick temporary patch.

File: src/mongo/MongoScrapbook.ts [hunk 2]
```diff
@@ -281,7 +281,9 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
...
+			this.addErrorToCommand(parseError(err), ctx);
```
Comment (StephenWeatherford) [inline src/mongo/MongoScrapbook.ts]: Why are you creating an error then parsing it and only using the message?

File: src/mongo/MongoScrapbook.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply src/mongo/MongoScrapbook.ts]: @StephenWeatherford I assumed that "patch" was the planned release we have for Cosmos DB next Wednesday (as a part of the app service "Connections" feature). So I had already gone ahead with the other approach: https://github.com/Microsoft/vscode-cosmosdb/pull/960

Comment (StephenWeatherford) [reply src/mongo/MongoScrapbook.ts]: I'd rather take enough time to make sure this is fixed properly after a quick patch.

Comment (PrashanthCorp) [reply src/mongo/MongoScrapbook.ts]: The screenshot shows errors thrown from functionCallContextToObject. Almost all the errors in the telemetry arrive from there. I've covered that approach below.

File: src/mongo/MongoScrapbook.ts (hunk 2 above)
Comment (PrashanthCorp) [reply src/mongo/MongoScrapbook.ts]: I didn't want to change a whole lot of code. addErrorToCommand takes in an IParsedError. I'll change the typing for addErrorToCommand to only require the a message field.

File: src/mongo/MongoScrapbook.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply src/mongo/MongoScrapbook.ts]: I still want an all-encompassing try/catch before I approve this. Something like this: https://github.com/Microsoft/vscode-cosmosdb/commit/869dd1dae9d51a9a381312adef2f3e7cc5e1b3d2?w=1

Comment (StephenWeatherford) [reply src/mongo/MongoScrapbook.ts]: That would be good.

File: src/mongo/MongoScrapbook.ts [hunk 3]
```diff
@@ -405,7 +411,7 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 		}
 	}
 
-	private addErrorToCommand(error: IParsedError, ctx: mongoParser.ArgumentContext | mongoParser.PropertyValueContext): void {
+	private addErrorToCommand(error: { message: string }, ctx: mongoParser.ArgumentContext | mongoParser.PropertyValueContext): void {
```
Comment (StephenWeatherford) [inline src/mongo/MongoScrapbook.ts]: Ah, I see.  Silly to create an error just to pull out them message, but don't change it for the patch.

File: src/mongo/MongoScrapbook.ts (hunk 1 above)
Comment (PrashanthCorp) [reply src/mongo/MongoScrapbook.ts]: Pushed. :)

**Pull request merged**